clang/include/clang/AST/Stmt.h

```diff
@@ -3193,7 +3193,7 @@ class AsmStmt : public Stmt {
   /// getOutputConstraint - Return the constraint string for the specified
   /// output operand.  All output constraints are known to be non-empty (either
   /// '=' or '+').
-  StringRef getOutputConstraint(unsigned i) const;
+  std::string getOutputConstraint(unsigned i) const;
```

Comment:
should these be llvm::SmallString?

Reply:
As discussed offline, std::string is better at interface boundaries https://llvm.org/docs/ProgrammersManual.html#string-like-containers